components/brave_wallet/browser/eth_data_parser.cc

```diff
@@ -6,224 +6,208 @@
+constexpr char kERC721SafeTransferFromSelector[] = "0x42842e0e";
+constexpr char kERC1155SafeTransferFromSelector[] = "0xf242432a";
+constexpr char kSellEthForTokenToUniswapV3Selector[] = "0x3598d8ab";
+constexpr char kSellTokenForEthToUniswapV3Selector[] = "0x803ba26d";
+constexpr char kSellTokenForTokenToUniswapV3Selector[] = "0x6af479b2";
+constexpr char kSellToUniswapSelector[] = "0xd9627aa4";
+constexpr char kTransformERC20Selector[] = "0x415565b0";
 
 }  // namespace
 
-bool GetTransactionInfoFromData(const std::string& data,
-                                mojom::TransactionType* tx_type,
-                                std::vector<std::string>* tx_params,
-                                std::vector<std::string>* tx_args) {
-  CHECK(tx_type);
-
-  if (tx_params)
-    tx_params->clear();
-  if (tx_args)
-    tx_args->clear();
-
-  *tx_type = mojom::TransactionType::Other;
-
-  static std::map<std::string, mojom::TransactionType> kEthDataFunctionHashes =
-      {{"0xa9059cbb", mojom::TransactionType::ERC20Transfer},
-       {"0x095ea7b3", mojom::TransactionType::ERC20Approve},
-       {"0x23b872dd", mojom::TransactionType::ERC721TransferFrom},
-       {"0x42842e0e", mojom::TransactionType::ERC721SafeTransferFrom},
-       {"0xf242432a", mojom::TransactionType::ERC1155SafeTransferFrom},
-       {"0x70a08231", mojom::TransactionType::Other}};
-
-  if (data.empty() || data == "0x0") {
-    *tx_type = mojom::TransactionType::ETHSend;
-    return true;
-  }
-  if (!IsValidHexString(data))
-    return false;
-  if (data.length() < 10) {
-    *tx_type = mojom::TransactionType::Other;
-    return true;
-  }
-
-  std::string function_hash = base::ToLowerASCII(data.substr(0, 10));
-  auto it = kEthDataFunctionHashes.find(function_hash);
-  if (it != kEthDataFunctionHashes.end())
-    *tx_type = it->second;
-  else
-    *tx_type = mojom::TransactionType::Other;
-
-  if (*tx_type == mojom::TransactionType::ERC20Transfer ||
-      *tx_type == mojom::TransactionType::ERC20Approve) {
-    std::string address, value;
-    std::string left_over_data = data.substr(10);
-    // Intentional copy of left_over_data
-    if (!GetAddressArgFromData(std::string(left_over_data), &address,
-                               &left_over_data))
-      return false;
-    if (!GetUint256HexFromData(std::string(left_over_data), &value,
-                               &left_over_data))
-      return false;
-    // Very strictly must have correct data
-    if (left_over_data.length() > 0)
-      return false;
-
-    if (tx_args) {
-      tx_args->push_back(address);
-      tx_args->push_back(value);
+absl::optional<std::tuple<mojom::TransactionType,     // tx_type
+                          std::vector<std::string>,   // tx_params
+                          std::vector<std::string>>>  // tx_args
+GetTransactionInfoFromData(const std::vector<uint8_t>& data) {
+  if (data.empty() || data == std::vector<uint8_t>{0x0})
+    return std::make_tuple(mojom::TransactionType::ETHSend,
```

Comment:
prefer wrapping in {} when 
```
if (...)
  do something
```
needs more than two lines

Reply:
Fixed a0185a70b0a7a7d5518792936e29ca187162aa43